interfaces/input_parameters/ring.py

```diff
@@ -182,182 +130,348 @@ class Ring:
+                                   "Particle object or a str.")
+
+        # Getting all sections and checking their types
+        if isinstance(Section_list, RingSection):
+            Section_list = [Section_list]
+        if not isinstance(Section_list, list):
+            raise excpt.InputError("The Section_list should be a list or " +
+                                   "a single RingSection object instance.")
+        else:
+            for section in Section_list:
+                if not isinstance(section, RingSection):
+                    raise excpt.InputError(
+                        "The Section_list should be exclusively composed " +
+                        "of RingSection object instances.")
+
+                # Setting the Ring as attribute of the sections for reference
+                section._Ring = self
+
+        self.Section_list = Section_list
+        self.n_sections = len(self.Section_list)
+
+        # Extracting the length of sections to get circumference
+        self.section_length_design = np.array(
+            [section.length_design for section in self.Section_list])
+        self.circumference = np.sum(self.section_length_design)
+
+        # Computing ring radius
+        self.radius = self.circumference / (2 * np.pi)
+
+        # Extracting the bending radius from all sections
+        self.bending_radius = np.array([
+            section.bending_radius for section in self.Section_list])
+
+        # Extracting the synchronous data from the sections
+        for index_section, section in enumerate(self.Section_list):
+            if index_section == 0:
+                self.synchronous_data = section.synchronous_data
+            else:
+                if (section.synchronous_data == self.synchronous_data).all:
+                    pass
+                else:
+                    warn_message = 'The synchronous data for all sections ' + \
+                        'is not identical. This case is not yet fully ' + \
+                        'tested and is under implementation.'
+                    warnings.warn(warn_message)
+
+        # Processing the momentum program
+        # Getting the options to get at which time samples the interpolation
+        # is made
         t_start = kwargs.pop('t_start', 0)
         t_stop = kwargs.pop('t_stop', np.inf)
         interp_time = kwargs.pop('interp_time', 0)
 
+        # Treating the input of the interp_time option
         if not hasattr(interp_time, '__iter__'):
             interp_time = (interp_time, )
 
+        # Getting the sample times
         sample_func, start, stop = tmng.time_from_sampling(*interp_time)
 
+        # Keeping only the values between the user defined bounds
         if t_start > start:
             start = t_start
         if t_stop < stop:
             stop = t_stop
 
-        synchronous_data.convert(self.Particle.mass, self.Particle.charge,
-                                 self.bending_radius)
```

Comment:
Is the conversion being done in the `section` objects now?

Reply:
No still in Ring, it was moved elsewhere and probably got eaten by the gruffalo in this commit 😅 , it's back in the more recent version